### ISSUE TITLE
[explicit-function-return-type] Private functions excluded?

### ISSUE THREAD

**Issue by josephzidell** [typescript-eslint/typescript-eslint#2862]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/explicit-function-return-type": ["error"]
  }
}
```

```TS
class Foo {
    private bar () {
        return 123;
    }
}
```

**Expected Result**
```
warning: Missing return type on function
```

**Actual Result**
It does not throw an error

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `^4.9.1` |
| `@typescript-eslint/parser`        | `^4.9.1` |
| `TypeScript`                       | `^4.1.2` |
| `ESLint`                           | `^7.15.0` |
| `node`                             | `v13.8.0` |

**Comment by bradzacher** [comment]
`explicit-function-return-type` simply reports on any and all functions.
This is not going to change.

If you're looking to limit it to public types only, consider instead using the `explicit-module-boundary-types` rule

**Comment by josephzidell** [comment]
@bradzacher Perhaps I wasn't clear, but it currently *is not reporting* on private functions

**Comment by josephzidell** [comment]
Nvm. My config was incorrect
